inline.go

```diff
@@ -212,6 +212,11 @@ func link(p *parser, out *bytes.Buffer, data []byte, offset int) int {
 	var t linkType
 	if offset > 0 && data[offset-1] == '!' {
 		t = linkImg
+		// if footnotes extension is ON and we've seen "![^", then it's not an
+		// image, it's a deferred footnote:
+		if p.flags&EXTENSION_FOOTNOTES != 0 && len(data)-1 > offset && data[offset+1] == '^' {
+			t = linkDeferredFootnote
+		}
```

Comment:
Behavior-wise this looks good to me. However, the current indentation style of the code makes this feel more like a hack/workaround than a long term solution (which it kinda is, since we're first setting `t` to `linkImg`, then over-writing it to be a `linkDeferredFootnote`, and also relying on a simple heuristic that most image URLs do not start with `^` as the first character).

Consider rewriting this if statement, which currently has 2 conditions `if offset > 0 && data[offset-1] == '!'` and `else if p.flags&EXTENSION_FOOTNOTES != 0`, into a switch statement with 3 conditions:

```Go
switch {
// ![^refId] == deferred footnote
case p.flags&EXTENSION_FOOTNOTES != 0 && len(data)-1 > offset && data[offset+1] == '^':
	// If footnotes extension is ON and we've seen "![^", then it's not an
	// image, it's a deferred footnote.
	t = linkDeferredFootnote

// ![alt] == image
case offset > 0 && data[offset-1] == '!':
	t = linkImg

// ^[text] == inline footnote
// [^refId] == deferred footnote
case p.flags&EXTENSION_FOOTNOTES != 0:
	if offset > 0 && data[offset-1] == '^' {
		t = linkInlineFootnote
	} else if len(data)-1 > offset && data[offset+1] == '^' {
		t = linkDeferredFootnote
	}
}
```

I think that would be more readable and maintainable. What do you think?